kafka/kafka.go

```diff
@@ -51,52 +51,58 @@ func NewKafkaPublisher() *kafkaPublisher {
 	return &kafkaPublisher{}
 }
 
+type MetricToPublish struct {
+	// The timestamp from when the metric was created.
+	Timestamp time.Time         `json:"timestamp"`
+	Namespace string            `json:"namespace"`
+	Data      interface{}       `json:"data"`
+	Unit      string            `json:"unit"`
+	Tags      map[string]string `json:"tags"`
+	Version_  int               `json:"version"`
+	// Last advertised time is the last time the snap agent was told about a metric.
+	LastAdvertisedTime time.Time `json:"last_advertised_time"`
+}
+
 // Publish sends data to a Kafka server
 func (k *kafkaPublisher) Publish(contentType string, content []byte, config map[string]ctypes.ConfigValue) error {
-
-	// Inserted and modified codes from intelsdi-x/snap-plugin-publisher-file/file/file.go
-	logger := log.New()
-	logger.Println("Publishing started")
-	var metrics []plugin.MetricType
+	var mts []plugin.MetricType
 
 	switch contentType {
 	case plugin.SnapGOBContentType:
 		dec := gob.NewDecoder(bytes.NewBuffer(content))
-		if err := dec.Decode(&metrics); err != nil {
-			logger.Printf("Error decoding: error=%v content=%v", err, content)
-			return err
+		// decode incoming metrics types
+		if err := dec.Decode(&mts); err != nil {
+			fmt.Fprintf(os.Stderr, "Error: invalid incoming content: %v, err=%v", content, err)
+			return fmt.Errorf("Cannot decode incoming content, err=%v", err)
 		}
 	default:
-		logger.Printf("Error unknown content type '%v'", contentType)
 		return fmt.Errorf("Unknown content type '%s'", contentType)
 	}
-
-	logger.Printf("publishing %v metrics to %v", len(metrics), config)
+	// format metrics types to metrics to be published
+	metrics := formatMetricTypes(mts)
```

Comment:
Please consider renaming func "formatMetricTypes" to "formatMetricTypesAsJson" - in future maybe we will support many formats simultaneously (with user selecting format via config).